src/wp-content/themes/twentytwentyone/functions.php

```diff
@@ -425,7 +425,7 @@ function twenty_twenty_one_scripts() {
 		get_template_directory_uri() . '/assets/js/polyfills.js',
 		array(),
 		wp_get_theme()->get( 'Version' ),
-		true
+		array( 'in_footer' => true )
```

Comment:
Isn't this also used in lots of other places and the other default themes? I'd argue we should either update all of these usages, or we keep those changes out of here (since the old way still works backward compatible anyway) and instead do it later in a separate follow up PR, to not make the review of this main PR here overly clunky.

Reply:
I generally think we should update these since passing true is the "old" way of specifying in-footer. Agree good to add in a separate PR though to reduce review burden.

Reply:
@johnbillion That was considered when designing this implementation, however it's worth noting that it is not a backward compatibility break. It's something that would need to be called out in the dev note: If you're using the new array format and would like to remain compatible with older WordPress versions (in case you use `$in_footer` as `false`), you will need to use the function conditionally.

That is something that has been required based on numerous WordPress core changes in the past, so I think it's worth considering as a trade-off. It's certainly not breaking anything, it would only be future code doing it wrong if there was any breakage. That said, I acknowledge this is not intuitive for a theme developer, so there is of course a downside to this.

Though any developer could easily solve this with something like:
```php
function my_enqueue_script( ..., array $args ) {
	if ( version_compare( get_bloginfo( 'version' ), '6.3.', '<' ) ) {
		wp_enqueue_script( ..., isset( $args['in_footer'] ) ? $args['in_footer'] : false );
	} else {
		wp_enqueue_script( ..., $args );
	}
}
```

Reply:
I feel we don't need to make any change here. The old usage `true` or new usage `array( 'in_footer' => true )` and; `false`  or new usage `array( 'in_footer' => false )` is exactly the same and if the args doesn't contain `in_footer` it defaults to true. Old usage is equally supported and works exactly the same way.

I feel we should keep it simple and not make any update if we are not introducing anything related to strategy here.

Reply:
> This seems like a change that makes the themes no longer compatible with versions of WordPress prior to this change. While array( 'in_footer' => true ) does evaluate to true and therefore works on older versions of WordPress, array( 'in_footer' => false ) still evaluates to true and would mean the script is incorrectly enqueued in the footer on older versions of WordPress.

Good point - this would break compatibility when these themes are used _in older versions of WordPress_ so we do need to be careful when applying these changes. 

> I would suggest we remove all the changes to various core scripts from this PR, as that makes code review of the actual main changes here unnecessarily clunky

I agree, lets handle those in a separate follow-up PR.

Reply:
> Though any developer could easily solve this with something like:

I just made a suggestion to eliminate the `script_args` key in https://github.com/WordPress/wordpress-develop/pull/4391/files#r1197166624

If we do that, then backwards-compatibility becomes much easier. If you want to support versions of WordPress older than 6.3, instead of having to do a `version_compare()` check:

```php
if ( version_compare( get_bloginfo( 'version' ), '6.3.', '<' ) ) {
	wp_enqueue_script( ..., isset( $args['in_footer'] ) ? $args['in_footer'] : false );
} else {
	wp_enqueue_script( ..., $args );
}
```

You can just do the enqueue in the original way, but then add a separate `wp_add_inline_data()` call to pass the `strategy` (which is ignored in WP<6.3):

```php
wp_enqueue_script( 'foo', 'foo.js', '1.0', array(), true /*in_footer*/ );
wp_add_inline_data( 'foo', 'strategy', 'defer' ); // Ignored in WP<6.3.
```

Once a theme/plugin is explicitly no longer supporting WP<6.3, they are free at any time to update the above code to:

```php
wp_enqueue_script( 'foo', 'foo.js', '1.0', array(), array( 'in_footer' => true, 'strategy' => 'defer' ) );
```

But this just becomes a shorthand for the former (which it is under the hood anyway).

Reply:
This is implemented in https://github.com/10up/wordpress-develop/pull/64